vizro-core/src/vizro/managers/_model_manager.py

```diff
@@ -84,28 +84,28 @@ def _get_models(
     def __get_model_children(self, model: Model) -> Generator[Model, None, None]:
         """Iterates through children of `model`.
 
-        Currently looks only through certain fields so might miss some children models.
+        Currently, this method looks only through certain fields (components, tabs, controls, actions, selector) and
+            their children so might miss some children models.
         """
         from vizro.models import VizroBaseModel
 
         if isinstance(model, VizroBaseModel):
             yield model
+        elif isinstance(model, list):
+            for single_model in model:
+                yield from self.__get_model_children(single_model)
+        elif isinstance(model, dict):
+            # We don't look through keys because Vizro models aren't hashable.
+            for single_model in model.values():
+                yield from self.__get_model_children(single_model)
```

Comment:
If we want to support searching through types like `tuple`, `set`.. we can adjust the implementation to the following one. (although I don't think it's necessary to do so now):

```suggestion
        from vizro.models import VizroBaseModel
        from collections.abc import Collection, Mapping

        if isinstance(model, VizroBaseModel):
            yield model
        elif isinstance(model, Collection) and not isinstance(model, str):
            for single_model in model:
                yield from self.__get_model_children(single_model)
        elif isinstance(model, Mapping):
            # We don't look through keys because Vizro models aren't hashable.
            for single_model in model.values():
                yield from self.__get_model_children(single_model)
```

Reply:
Nice, it looks good! I'm tempted to make this change right away. I guess we might as well because I don't think it will cause any problems? But this model manager code will change in the nearish future anyway, and there will hopefully be a better way to iterate through composite model field types (list, tuple, etc.). So up to you.

Reply:
Did it -> https://github.com/mckinsey/vizro/pull/929/commits/4f5e717f7cf8fb9f82e1de81d63e49af73c5aee9 😄 

I added tests for `tuple`, and as vm.Filter is not hashable I wasn't able to add tests for `sets`.
Also, I removed the following test cases as I think are irrelevant: (they pass but I think are irrelevant)
```
            # model nested in a list of dictionaries
            (lambda x: [{"key": x}]),
            # model nested in a dictionary of lists
            (lambda x: {"key": [x]}),
```